### PR TITLE
Move compute mirror logic into async.

### PR DESCRIPTION
**Pull request by dbatiste**
@dlockhart : this change wraps the compute mirror logic in `this.async` which will ensure that child components (in this case `iron-icon`) are ready when this logic runs.  There is no guarantee that the order components are `ready` in a particular order.

Background: I ran into timing issue where the `svg` was not distributed yet when this logic was run.  Wrapping in this component's `async` resolves the issue.

### CODE REVIEW

**Comment by dbatiste** [comment]
Following our conversation, I am going to update to handle changes to the icon as well, which affects this change a little bit... so more change coming.

File: d2l-icon.html [hunk 1]
```diff
@@ -33,10 +33,12 @@
...
-				var mirrorRtl = this.__computeMirrorRtl();
-				if (mirrorRtl) {
-					this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
-				}
+				this.async(function() {
+					var mirrorRtl = this.__computeMirrorRtl();
+					if (mirrorRtl) {
+						this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
+					}
+				});
```
Comment (dlockhart) [inline d2l-icon.html]: A little surprised this works without binding the async callback to `this`...

**Comment by dbatiste** [comment]
I've updated things a bit here:
* added support for changing `icon` attr on `d2l-icon`
* added some tests (they know a bit about internals to enable testing, but i think worthwhile)
* updated demo page

File: d2l-icon.html [hunk 2]
```diff
@@ -64,7 +63,32 @@
...
+				if (!svg) {
+					return false;
+				}
+				/* IE prevent svg from being focusable */
+				svg.setAttribute('focusable', 'false');
+				var mirrorRtl = this.__computeMirrorRtl(svg);
+				if (mirrorRtl) {
+					this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
+				} else {
+					this.$$('iron-icon').removeAttribute('data-mirror-rtl');
```
Comment (dbatiste) [inline d2l-icon.html]: It was necessary to add this `removeAttribute` call because the same `iron-icon` instance is used.

File: d2l-icon.html [hunk 3]
```diff
@@ -64,7 +63,32 @@
...
+			__initialize: function() {
+				var svg = this.__findSvg(this);
+				if (!svg) {
+					return false;
+				}
+				/* IE prevent svg from being focusable */
+				svg.setAttribute('focusable', 'false');
+				var mirrorRtl = this.__computeMirrorRtl(svg);
+				if (mirrorRtl) {
+					this.$$('iron-icon').setAttribute('data-mirror-rtl', '');
```
Comment (dlockhart) [inline d2l-icon.html]: Should `__getIronIcon()` be called here or can it just be removed?

**Comment by dlockhart** [comment]
Looks great @dbatiste, thanks for fixing this up and adding the tests! Next: Galen perceptual diffs...

File: d2l-icon.html (hunk 3 above)
Comment (dbatiste) [inline d2l-icon.html]: Ah good eye!  So, I created this helper just to facilitate the unit tests.

Comment (dbatiste) [inline d2l-icon.html]: Ideally I would not need such a helper.  A perceptual diff would enable testing the flip case better.  :)

**Pull request merged**